src/htr2hpc/train/run.py

```diff
@@ -22,11 +25,141 @@
 
 # kraken python package provides paths to the default best models for both modes
 default_model = {
-    "segmentation": SEGMENTATION_DEFAULT_MODEL,
-    "transcription": DEFAULT_MODEL,
+    "Segment": SEGMENTATION_DEFAULT_MODEL,
+    "Recognize": DEFAULT_MODEL,
 }
 
 
+@dataclass
+class TrainingManager:
+    base_url: str
+    api_token: str
+    work_dir: pathlib.Path
+    document_id: int
+    training_mode: str
+    model_name: str
+    num_workers: int
+    parts: Optional[intspan] = None
+    model_id: Optional[int] = None
+    existing_data: bool = False
+    show_progress: bool = True
+
+    def __post_init__(self):
+        # initialize api client
+        self.api = eScriptoriumAPIClient(self.base_url, self.api_token)
+        # TODO : check api access works before going too far?
+        # (currently does not handle connection error very gracefully)
+
+        # store the path to original working directory before changing directory
+        self.orig_working_dir = pathlib.Path.cwd()
+
+    def training_prep(self):
+        # create necessary directories and download training data and model file
+
+        self.training_data_dir = self.work_dir / "parts"
+        if not self.existing_data:
+            self.training_data_dir.mkdir()
+        get_training_data(
+            self.api, self.training_data_dir, self.document_id, self.parts
+        )
+
+        # if model id is specified, download the model from escriptorium API,
+        # confirming that it is the appropriate type (segmentation/transcription)
+        # NOTE: currently ignores existing data flag, since we need model file name
+        if self.model_id:
+            self.model_file = get_model(
+                self.api,
+                self.model_id,
+                self.training_mode,
+                self.work_dir,
+            )
+        # if model id is not specified, use the default from kraken
+        else:
+            # get the appropriate model file for the requested training mode
+            # kraken default defs are path objects
+            self.model_file = default_model[self.training_mode]
+
+        # create a directory and path for the output model file
+        self.output_model_dir = self.work_dir / "output_model"
+        # remove the output model directory to avoid confusion with any old
+        # model files from a previous run
+        if self.existing_data:
+            rmtree(self.output_model_dir)
+        self.output_model_dir.mkdir()
+        self.output_modelfile = self.output_model_dir / self.model_name
+
+    def monitor_slurm_job(self, job_id):
+        # get initial job status (typically PENDING)
+        job_status = slurm_job_queue_status(job_id)
+        # typical states are PENDING, RUNNING, SUSPENDED, COMPLETING, and COMPLETED.
+        # https://slurm.schedmd.com/job_state_codes.html
+        # end states could be FAILED, CANCELLED, OUT_OF_MEMORY, TIMEOUT
+        # * but note that squeue only reports on pending & running jobs
```

Comment:
Not a requirement, but since we have access to this information we could consider creating / hooking into a task report instance for the training job and passing certain relevant states back into eScriptorium. (We will probably come back to this point when we get to connecting the SSH integration we have here to the eScriptorium GUI...)